Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionURLSchemeHandlerCocoa.mm

```diff
@@ -104,8 +104,10 @@
             loadingExtensionMainFrame = true;
         }
 
-        auto *fileData = extensionContext->extension().resourceDataForPath(requestURL.path().toString());
+        NSError *error;
+        auto *fileData = extensionContext->extension().resourceDataForPath(requestURL.path().toString(), &error);
         if (!fileData) {
+            extensionContext->recordError(error);
```

Comment:
I wonder if you should be more defensive about not calling `recordError()` here if `error` is nil? Alternately, you could make `recordError()` more robust in this case by returning early instead of just asserting that the error parameter isn't nil like you currently do. I'm not sure what WebKit's convention is here, I know in general we try to avoid unnecessary checks where possible, so maybe it's OK to assume that `resourceDataForPath()` will always set `error` on failure.

Reply:
An error is expected here, if data is `nil`. I did add `recordErrorIfNeeded()` in this PR that does do a check for `nil` first for the cases where it is needed.